src/ripple/app/misc/NetworkOPs.cpp

```diff
@@ -3117,6 +3286,296 @@ NetworkOPsImp::unsubAccountInternal(
+                                    return true;
+                                }
+                            }
+                        }
+                    }
+                    catch (std::exception const&)
+                    {
+                        continue;
+                    }
+                }
+
+                return false;
+            };
+
+            auto send = [&](Json::Value const& jvObj,
+                            bool unsubscribe) -> bool {
+                if (auto sptr = subInfo.sinkWptr_.lock(); sptr)
+                {
+                    sptr->send(jvObj, true);
+                    if (unsubscribe)
+                        unsubAccountHistory(sptr, accountId, false);
+                    return true;
+                }
+                else
+                {
+                    return false;
+                }
+            };
+
+            /*
+             * search backward until the genesis ledger or asked to stop
+             */
+            while (lastLedgerSeq >= 2 && !subInfo.index_->stopHistorical_)
+            {
+                auto startLedgerSeq =
+                    (lastLedgerSeq > 256 + 2 ? lastLedgerSeq - 256 : 2);
+                JLOG(m_journal.trace())
+                    << "AccountHistory job for account " << toBase58(accountId)
+                    << ", working on ledger range [" << startLedgerSeq << ","
+                    << lastLedgerSeq << "]";
+
+                auto haveRange = [&]() -> bool {
+                    std::uint32_t validatedMin = UINT_MAX;
+                    std::uint32_t validatedMax = 0;
+                    auto haveSomeValidatedLedgers =
+                        app_.getLedgerMaster().getValidatedRange(
+                            validatedMin, validatedMax);
+
+                    return haveSomeValidatedLedgers &&
+                        validatedMin <= startLedgerSeq &&
+                        lastLedgerSeq <= validatedMax;
+                }();
+
+                if (!haveRange)
+                {
+                    JLOG(m_journal.debug())
+                        << "AccountHistory reschedule job for account "
+                        << toBase58(accountId) << ", incomplete ledger range ["
+                        << startLedgerSeq << "," << lastLedgerSeq << "]";
+                    setAccountHistoryJobTimer(subInfo);
+                    return;
+                }
+
+                RelationalDBInterface::AccountTxPageOptions options = {
+                    accountId, startLedgerSeq, lastLedgerSeq, {}, 0, true};
+
+                auto [txns, marker] =
+                    dynamic_cast<RelationalDBInterfaceSqlite*>(
+                        &app_.getRelationalDBInterface())
+                        ->newestAccountTxPage(options);
+
+                for (auto const& [tx, meta] : txns)
+                {
+                    if (!tx || !meta)
+                    {
+                        send(rpcError(rpcINTERNAL), true);
```

Comment:
I think we should log and assert here.

Reply:
Do `tx` and `meta` guaranteed to be not null? I cannot tell from the `getRelationalDBInterface()` function document, and too unfamiliar with database code.

Reply:
Well actually, in the regular `account_tx` handler, we do check for null too, so I think it's fine to leave this as is.